src/server.ts

```diff
@@ -49,7 +53,7 @@ new ItemRouter(app).mount()
 new CollectionRouter(app).mount()
 new CommitteeRouter(app).mount()
 new RarityRouter(app).mount()
-new ForumRouter(app).mount()
+new ForumRouter(app, metrics).mount()
```

Comment:
Not sure if it is possible but maybe we can avoid override the Router's constructor and send `metrics` at `mount`

Reply:
The reason behind not doing that is that we want to set an instance of the metric component once that can be used by every route handler defined in the RouterClass.
There are three alternatives:
- We set it in the constructor once and use it in all the routers or methods in the class.
- We pass it to the mount method and propagate it to each of the routes being created by doing something like: `server.handleRequest((req, res) => this.post(req, res, metrics))`
- We pass it to the mount method and set it on each mount, making the metrics mutable, so any handler or method in the class can use it (I wouldn't to this, as metrics isn't meant to be mutable)

Reply:
Great! thanks for the explanation. Another alternative is to create and use a middleware to inject the metrics like `withMetrics` and use it like

```ts
this.router.post(
      '/collections/:id/post',
      withAuthentication,
      withCollectionExists,
      withCollectionAuthorization,
      withMetrics,
      server.handleRequest(this.post)
    )
```

I will leave this to you & @nicosantangelo.

Reply:
Because we're borrowing the metrics from well-known-components, I think it makes sense to have this as an instance of our routers. This is because it's a complex object being created once on the `server.ts` and re-used later.

That said, what do you think of generalizing this to make our Routers metric aware? I'm thinking we can move the logic of accepting a `metricsComponent` to the abstract class and that we could type the strings that make our metrics like this:

```ts
export enum MetricKeys {
  FORUM_POST_FAILED = 'dcl_published_collection_forum_post_failed',
  FORUM_POST_ALREADY_EXISTS = 'dcl_published_collection_forum_post_already_exists',
}

export const metricDeclarations = {
  [MetricKeys.FORUM_POST_FAILED]: {
    (...)
```

and then either have:

```ts
export class Router<M extends string = string> {
  protected router: express.Router
  protected metrics?: IMetricsComponent<M>
  (...)

// or

export class MetricsRouter<M extends string = string> extends Router {
  protected metrics?: IMetricsComponent<M>
```

What do you think @LautaroPetaccio 

Lastly, but importantly, I think the increment on ForumRouter, that's using `'dcl_published_collection_forum_post_already_exists'` it's not correctly typed, as it's missing from the `IMetricsComponent` generic definition

Reply:
Done! I've added the metrics property as an optional value in the Router.